lib/src/widgets/calendar_page.dart

```diff
@@ -6,31 +6,62 @@ import 'package:flutter/widgets.dart';
 class CalendarPage extends StatelessWidget {
   final Widget Function(BuildContext context, DateTime day)? dowBuilder;
   final Widget Function(BuildContext context, DateTime day) dayBuilder;
+  final Widget Function(BuildContext context, DateTime day)? weekNumberBuilder;
   final List<DateTime> visibleDays;
   final Decoration? dowDecoration;
   final Decoration? rowDecoration;
   final bool dowVisible;
+  final bool weekNumberVisible;
+  final double? dowHeight;
 
   const CalendarPage({
     Key? key,
     required this.visibleDays,
     this.dowBuilder,
     required this.dayBuilder,
+    this.weekNumberBuilder,
     this.dowDecoration,
     this.rowDecoration,
     this.dowVisible = true,
+    this.weekNumberVisible = false,
+    this.dowHeight,
   })  : assert(!dowVisible || dowBuilder != null),
         super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return Table(
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
       children: [
-        if (dowVisible) _buildDaysOfWeek(context),
-        ..._buildCalendarDays(context),
+        if (weekNumberVisible)
+          _buildWeekNumbers(context),
+        Expanded(
+          child: Table(
+            children: [
+              if (dowVisible) _buildDaysOfWeek(context),
+              ..._buildCalendarDays(context),
+            ],
+          ),
+        ),
       ],
     );
   }
+  
+  Widget _buildWeekNumbers(BuildContext context) {
+    final rowAmount = visibleDays.length ~/ 7;
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 4.0),
```

Comment:
Let's move this padding into null `weekNumberBuilder` implementation in table_calendar.dart file. This will allow using the actual `weekNumberBuilder` without the inherent 4.0 horizontal padding.